raiden/smart_contracts/ChannelManagerLibrary.sol

```diff
@@ -37,6 +38,28 @@ library ChannelManagerLibrary {
     /// @return channel (address) the address of the NettingChannelContract of the two parties.
     function getChannelWith(Data storage self, address caller_address, address partner)
         constant
+        returns (address, bool)
+    {
+        bytes32 party_hash = partyHash(caller_address, partner);
+        if (self.channel_addresses[party_hash] == 0x0 || 
+            !contractExists(self, self.channel_addresses[party_hash])) {
```

Comment:
Well. Right now this would happen whenever a contract has been settled. We `kill` the contract after it's been settled. But since there isn't really a way of notifying the ChannelManager of this (as far as I know), we still have an entry in the mapping until a new channel is opened between those two parties. It's not a solution I like, but haven't been able to come up with any other way of doing this.